ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -528,10 +536,80 @@ protected void initTreeTable() {
     doInit();
   }
 
+  private static Comparator<Task> beginComparatorAsc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return t1.getStart().compareTo(t2.getStart());
+    }
+  };
+
+  private static Comparator<Task> beginComparatorDesc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return -t1.getStart().compareTo(t2.getStart());
+    }
+  };
+
+  private static Comparator<Task> endComparatorAsc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return t1.getEnd().compareTo(t2.getEnd());
+    }
+  };
+
+  private static Comparator<Task> endComparatorDesc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return -t1.getEnd().compareTo(t2.getEnd());
+    }
+  };
+  
   protected void doInit() {
     setRootVisible(false);
     myCustomPropertyManager.addListener(this);
 
+    getTree().getTableHeader().addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent mouseEvent) {
+        myProject.getTaskManager().getTaskHierarchy().sort(new Comparator<Task>() {
+          @Override
+          public int compare(Task t1, Task t2) {
+            return t1.getStart().compareTo(t2.getStart());
+          }
+        });
+      }
+    });
+
+    getTree().getTableHeader().addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent mouseEvent) {
+        int index = getTable().columnAtPoint(mouseEvent.getPoint());
+        ColumnImpl column = myTableHeaderFacade.findColumnByViewIndex(index);
+
+
+        if (column.myStub.getID().equals(TaskDefaultColumn.BEGIN_DATE.getStub().getID())) {
+          if (column.getSort() > 0) {
+            column.setSort(-1);
+            myProject.getTaskManager().getTaskHierarchy().sort(beginComparatorDesc);
```

Comment:
and if comparator was in the TaskDefaultColumn you could just write here `column.myStub.getComparator(column.getSort())`